delivery-status-service/src/main/java/com/tracking_delivery_system/delivery_status_service/controller/DeliverController.java

```diff
@@ -0,0 +1,39 @@
+package com.tracking_delivery_system.delivery_status_service.controller;
+
+import com.tracking_delivery_system.delivery_status_service.dto.NewDeliverDto;
+import com.tracking_delivery_system.delivery_status_service.model.Deliver;
+import com.tracking_delivery_system.delivery_status_service.model.StatusUpdate;
+import com.tracking_delivery_system.delivery_status_service.repository.DeliverRepository;
+import com.tracking_delivery_system.delivery_status_service.service.KafkaProducer;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/delivery")
+@RequiredArgsConstructor
+public class DeliverController {
+    @Autowired
```

Comment:
If you are using @RequiredArgsConstructor it means Lombok will create constructor for each service you have as a parameter in your class. And If you are using @Autowired - you provide dependency injection using Spring DI. Yu you have to you only one of this cases but not both